docs/lang/articles/basic/syntax.md

```diff
@@ -59,26 +81,30 @@ my_kernel(24, 3.2)  # prints: 27.2
 ```
 
 :::note
+Taichi supports scalars, `ti.Matrix` and`ti.Vector` as kernel arguments.
```

Comment:
For `ti.Matrix` and `ti.Vector` , now we only pass it by value. It is different from python `list` and c++ `array`. Maybe we should mention it?

Reply:
IIRC we support passing `ti.Matrix` and `ti.Vector` by reference by adding type annotation `ti.template()`, and anything passed by reference is inlined into the kernel during transformation. So, maybe we needn't mention it specifically because it is already stated in https://docs.taichi.graphics/lang/articles/advanced/meta#template-metaprogramming ?

Reply:
Regarding this, I think we need to mention that passing Python objects into kernels with `ti.template()` are read-only.

Reply:
> Regarding this, I think we need to mention that passing Python objects into kernels with `ti.template()` are read-only because it is basically instantiating templates.

In the current implementation, it seems we allow redefinitions of `ti.template` arguments once because the template arguments are placed in the global variable dict. The previous variable is shadowed by the new variable created which is in the variable dict of transformer. Should we ban this?

Reply:
Yes we should ban this.